api/src/main/java/bio/terra/tanagra/service/underlay/UnderlayConversion.java

```diff
@@ -406,16 +408,47 @@ public static Underlay convert(bio.terra.tanagra.proto.underlay.Underlay underla
             convert(hierarchyProto.getChildrenTable().getTableFilter(), columns));
       }
 
+      Column node =
+          retrieve(hierarchyProto.getPathsTable().getNode(), columns, "node", hierarchyProto);
+      Column path =
+          retrieve(hierarchyProto.getPathsTable().getPath(), columns, "path", hierarchyProto);
+      Hierarchy.PathsTable.Builder pathsTable =
+          Hierarchy.PathsTable.builder().node(node).path(path);
+
       Hierarchy hierarchy =
           Hierarchy.builder()
               .descendantsTable(descendantsTable.build())
               .childrenTable(childrenTable.build())
+              .pathsTable(pathsTable.build())
               .build();
+      // add a new attribute+mapping to the entity that contains the hierarchy path
+      Attribute hierarchyPathAttribute =
+          Attribute.builder()
+              .name("t_path_" + attribute.name())
```

Comment:
nit: should we validate user-specified attribute names in general against reserved Tanagra generated prefixes? e.g. I imagine things crash and burn if our condition entity already had an attribute with this name, however unlikely.

Reply:
Yes, good point. I added an `isGenerated` property to the internal `Attribute` class, so we can validate that the name doesn't include the prefix for user-defined, not-generated properties. Per your other comment, I'm thinking we can also use this property in the future to determine whether an attribute should be selectable by a user in dataset builder.